Storage/Storage/Bookmarks.swift

```diff
@@ -34,14 +36,11 @@ public struct BookmarkRoots {
 /**
  * The immutable base interface for bookmarks and folders.
  */
-@objc public protocol BookmarkNode {
-    var guid: String { get }
-    var title: String { get }
-    var icon: UIImage { get }
-}
-
-@objc public protocol Bookmark : BookmarkNode {
-    var url: String! { get }
+public class BookmarkNode {
+    public var id: Int? = nil
+    public var guid: String = ""
```

Comment:
To be clear, this is from old code. I agree these guids should be optional (and private for now since no one outside of Storage needs them, although that will probably change). Storage can fill them in.

BUT, for now I just forced them to be initialized in the constructor as we had done before. I'll file a bug to make them optional. I'm not sure about titles. It seems like a bookmark without a title (even an empty one) is broken.